### ISSUE TITLE
Parsing issues when option has separator attribute and when empty value provided... fails

### ISSUE THREAD

**Issue by hrnagesh-zz** [commandlineparser/commandline#420]
```
 [Option('b', "BookIds", Required = false, HelpText = "List of Books", Separator = ',')]
        public IEnumerable<string> Bookids { get; set; }

[Option('v', "Verbose", Required = false, Hidden = true, HelpText = "Generate Book details", Default = "Yes")]
        public string All { get; set; }
```
ProcessBooks -b -v NO

------------
If option has separator attribute and when empty value provided... fails; it considers other argument as its values..

In this case; -b has value -v and NO;

**Comment by lobster2012-user** [comment]
my fix
```csharp

public static IEnumerable<Token> Partition(
            IEnumerable<Token> tokens,
            Func<string, Maybe<TypeDescriptor>> typeLookup)
        {
            return from tseq in tokens.Pairwise(
                (f, s) =>
                        f.IsName() /*&& s.IsValue() fix*/

```

```csharp

public static Result<IEnumerable<Token>, Error> ExplodeOptionList(
            Result<IEnumerable<Token>, Error> tokenizerResult,
            Func<string, Maybe<char>> optionSequenceWithSeparatorLookup)
        {
            var tokens = tokenizerResult.SucceededWith().Memorize();

            var replaces = tokens.Select((t, i) =>
                optionSequenceWithSeparatorLookup(t.Text)
                    .MapValueOrDefault(sep => Tuple.Create(i + 1, sep),
                        Tuple.Create(-1, '\0'))).SkipWhile(x => x.Item1 < 0).Memorize();

            var exploded = tokens.Select((t, i) =>
            {
                return replaces.FirstOrDefault(x =>
                {
                    return x.Item1 == i && t.IsValue();//fix
                }).ToMaybe()

```

**Comment by wo-roberts** [comment]
Separately found and fixed this issue and happy to put a PR in if you're happy?

**Comment by rmunn** [comment]
This will be fixed by #684 once it's merged in. (It would also be fixed by #672, but #672 misses a different bug in the same code, so #684's version of the fix is better).
